src/main/scala/metadata.scala

```diff
@@ -0,0 +1,83 @@
+package suffuse
+
+/** AKey is an attribute key: a Metadata map key which carries
+ *  both the type of the corresponding value, and a value to
+ *  provide when there is no value in the map. This as opposed
+ *  to wrapping everything in sight in Option.
+ */
+final class AKey[A](description: String, empty: => A) {
+  def emptyValue: A = empty
```

Comment:
Also, the pieces of the Metadata class work as a team. If you feel strongly enough about it you should try rewriting it without Empty, and I think the collateral damage done to the interface will be clear even before we look beyond that file.